dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -1626,3 +1627,152 @@ HRESULT WINAPI CFSFolder::CallBack(IShellFolder *psf, HWND hwndOwner, IDataObjec
 
     return Shell_DefaultContextMenuCallBack(this, pdtobj);
 }
+
+static HBITMAP DoLoadPicture(LPCWSTR pszFileName)
+{
+    // open the picture file
+    HANDLE hFile;
+    hFile = CreateFileW(pszFileName, GENERIC_READ, FILE_SHARE_READ,
+                        NULL, OPEN_EXISTING, 0, NULL);
+    if (hFile == INVALID_HANDLE_VALUE)
+        return NULL;
+
+    // get the file size
+    DWORD cbGlobal = GetFileSize(hFile, NULL);
+    if (cbGlobal == INVALID_FILE_SIZE)
+    {
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    // allocate
+    HGLOBAL hGlobal = GlobalAlloc(GMEM_MOVEABLE, cbGlobal);
+    if (hGlobal == NULL)
+    {
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    // read it
+    LPVOID pvGlobal = GlobalLock(hGlobal);
+    DWORD cbRead;
+    if (!pvGlobal || !ReadFile(hFile, pvGlobal, cbGlobal, &cbRead, NULL) ||
+        cbRead != cbGlobal)
+    {
+        GlobalUnlock(hGlobal);
+        GlobalFree(hGlobal);
+        CloseHandle(hFile);
+        return NULL;
+    }
+
+    GlobalUnlock(hGlobal);
+
+    // close the file
+    CloseHandle(hFile);
+
+    // load the picture
+
+    HBITMAP hbm = NULL;
+    IPicture *pPicture = NULL;
+    IStream *pStream = NULL;
+    if (CreateStreamOnHGlobal(hGlobal, TRUE, &pStream) == S_OK)
```

Comment:
Might be simpler to use `SHCreateStreamOnFileEx` ?

Reply:
OK.